packages/@icon-magic/config-reader/src/schemas/config-schema.ts

```diff
@@ -3,14 +3,61 @@ const variantProperties = {
   path: { type: 'string', minLength: 1 }
 };
 
+const pluginProperties = {
+  name: { type: 'string' },
+  iterants: {
+    type: ['array'],
+    items: {
+      type: 'string'
+    }
+  },
+  params: ['object'],
+  writeToOutput: {
+    type: 'boolean',
+    default: true
+  }
+};
+
 const buildConfigProperties = {
-  outputPath: { type: ['string', 'null'] },
-  plugins: { type: ['array', 'null'] }
+  outputPath: {
+    type: ['string', 'null'],
+    minLength: 1
```

Comment:
minLength for the string or null?

Reply:
Only applies to the string. Restrictions (like minLength) are only checked when the document is of a type that is compatible with the keyword. https://cswr.github.io/JsonSchema/spec/multiple_types/

minLength is not a compatible keyword for type null.